lib/core/CMakeLists.txt

```diff
@@ -55,22 +197,75 @@ add_library(
   "${CMAKE_CURRENT_SOURCE_DIR}/src/json_events.cpp"
   "${CMAKE_CURRENT_SOURCE_DIR}/src/key_value_proxy.cpp"
   "${CMAKE_CURRENT_SOURCE_DIR}/src/list.cpp"
-  "${CMAKE_CURRENT_SOURCE_DIR}/src/msParam.cpp"
-  "${CMAKE_CURRENT_SOURCE_DIR}/src/obf.cpp"
-  "${CMAKE_CURRENT_SOURCE_DIR}/src/osauth.cpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/msParam.cpp" # #includes headers from other places
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/obf.cpp" # #includes headers from other places
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/osauth.cpp" # #includes headers from other places
   "${CMAKE_CURRENT_SOURCE_DIR}/src/packStruct.cpp"
-  "${CMAKE_CURRENT_SOURCE_DIR}/src/parseCommandLine.cpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/parseCommandLine.cpp" # #includes irods/rods.h
   "${CMAKE_CURRENT_SOURCE_DIR}/src/process_stash.cpp"
-  "${CMAKE_CURRENT_SOURCE_DIR}/src/rcGlobal.cpp"
-  "${CMAKE_CURRENT_SOURCE_DIR}/src/rcMisc.cpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/rcGlobal.cpp" # #includes irods/rods.h
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/rcMisc.cpp" # #includes headers from other places
   "${CMAKE_CURRENT_SOURCE_DIR}/src/region.cpp"
   "${CMAKE_CURRENT_SOURCE_DIR}/src/rodsError.cpp"
-  "${CMAKE_CURRENT_SOURCE_DIR}/src/rodsLog.cpp"
-  "${CMAKE_CURRENT_SOURCE_DIR}/src/rodsPath.cpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/rodsLog.cpp" # #includes irods/rods.h
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/rodsPath.cpp" # #includes headers from other places
   "${CMAKE_CURRENT_SOURCE_DIR}/src/stringOpr.cpp"
   "${CMAKE_CURRENT_SOURCE_DIR}/src/system_error.cpp"
   "${CMAKE_CURRENT_SOURCE_DIR}/src/version.cpp"
 )
+
+set(
+  IRODS_LIB_CORE_HEADERS_COMMON_GENERATORS
```

Comment:
Noticed there's no `_PUBLIC` on this one. Is this is a useful distinction for the generators, or no?

Reply:
Generators are always private regardless of the private/public status of the headers they produce.

Reply:
Got it, thanks.